### PR TITLE
Problem: evm transfer is not efficient

### PR DESCRIPTION
**Pull request by yihuang**
## Description

Solution:
- override the Transfer callback in block context to do transfer directly, instead of doing burning and minting.

This makes https://github.com/crypto-org-chain/cosmos-sdk/pull/239 unnecessary, will revert later.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -43,7 +43,11 @@ Ref: https://keepachangelog.com/en/1.0.0/
 - (feemarket) [#433](https://github.com/crypto-org-chain/ethermint/pull/433) Fix sdk int conversion panic with baseFee.
 * (rpc) [#434](https://github.com/crypto-org-chain/ethermint/pull/434) No need gasPrice when patch gasUsed for `eth_getTransactionReceipt`.
 * (rpc) [#439](https://github.com/crypto-org-chain/ethermint/pull/439), [#441](https://github.com/crypto-org-chain/ethermint/pull/441) Align trace response for failed tx with go-ethereum.
+
+### Improvements
+
 * (statedb) [#446](https://github.com/crypto-org-chain/ethermint/pull/446) Re-use the cache store implementation with sdk.
+* (evm) [#]() Refactor the evm transfer to be more efficient.
```
Comment (yihuang) [inline CHANGELOG.md]:
```suggestion
* (evm) [#448](https://github.com/crypto-org-chain/ethermint/pull/448) Refactor the evm transfer to be more efficient.
```

File: go.mod [hunk 2]
```diff
@@ -247,7 +247,7 @@ replace (
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.2.0
 	github.com/cockroachdb/pebble => github.com/cockroachdb/pebble v0.0.0-20230209160836-829675f94811
 	// release/v0.50.x
-	github.com/cosmos/cosmos-sdk => github.com/crypto-org-chain/cosmos-sdk v0.46.0-beta2.0.20240402015425-ec314e8e2d07
+	github.com/cosmos/cosmos-sdk => github.com/yihuang/cosmos-sdk v0.43.0-beta1.0.20240402060507-dd2eed16118e
```
Comment (yihuang) [inline go.mod]: https://github.com/crypto-org-chain/cosmos-sdk/pull/247
make sure it still works after the sdk revert

File: x/evm/statedb/statedb.go [hunk 3]
```diff
@@ -382,6 +386,22 @@ func (s *StateDB) Context() sdk.Context {
  * SETTERS
  */
 
+// Transfer from one account to another
+func (s *StateDB) Transfer(sender, recipient common.Address, amount *big.Int) {
+	if amount.Sign() == 0 {
```
Comment (mmsqe) [inline x/evm/statedb/statedb.go:391]: should we align with [other](https://github.com/crypto-org-chain/ethermint/blob/a76e99d87ddbc9ff75e132f7d0b8efc92b820072/x/evm/statedb/statedb.go#L407) check

Comment (yihuang) [reply x/evm/statedb/statedb.go:391]: maybe we should do a panic if it's negative

Comment (yihuang) [reply x/evm/statedb/statedb.go:391]: done

**Review by mmsqe — APPROVED** [review]

**Pull request merged**